src/main/java/org/opensearch/flowframework/workflow/HttpHostStep.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.hc.core5.http.HttpHost;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.support.PlainActionFuture;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.util.ParseUtils;
+
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static org.opensearch.flowframework.common.CommonValue.HOSTNAME_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.HTTP_HOST_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.PORT_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.SCHEME_FIELD;
+
+/**
+ * Step to register parameters for an HTTP Connection to a Host
+ */
+public class HttpHostStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(HttpHostStep.class);
+    PlainActionFuture<WorkflowData> hostFuture = PlainActionFuture.newFuture();
+    static final String NAME = "http_host";
+
+    @Override
+    public PlainActionFuture<WorkflowData> execute(
+        String currentNodeId,
+        WorkflowData currentNodeInputs,
+        Map<String, WorkflowData> outputs,
+        Map<String, String> previousNodeInputs
+    ) {
+        Set<String> requiredKeys = Set.of(SCHEME_FIELD, HOSTNAME_FIELD, PORT_FIELD);
+        Set<String> optionalKeys = Collections.emptySet();
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs
+            );
+
+            String scheme = validScheme(inputs.get(SCHEME_FIELD));
+            String hostname = inputs.get(HOSTNAME_FIELD).toString();
+            int port = validPort(inputs.get(PORT_FIELD));
+
+            HttpHost httpHost = new HttpHost(scheme, hostname, port);
+
+            hostFuture.onResponse(
+                new WorkflowData(
+                    Map.ofEntries(Map.entry(HTTP_HOST_FIELD, httpHost)),
+                    currentNodeInputs.getWorkflowId(),
+                    currentNodeInputs.getNodeId()
+                )
+            );
+
+            logger.info("Http Host registered successfully {}", httpHost);
+
+        } catch (FlowFrameworkException e) {
+            hostFuture.onFailure(e);
+        }
+        return hostFuture;
+    }
+
+    private String validScheme(Object o) {
```

Comment:
We might need to add more validation here based on the cluster settings potentially? It depends how we want the behavior to be but for example if `plugins.security.ssl.http.enabled: true` OS only allows https, we might want to also follow that here by only allowing https. Could be other cases on FGAC where this would be better too.

Reply:
For calls to OpenSearch itself, the user won't be providing the http host, so this whole step won't apply.  This is only for external APIs.

Reply:
How will we do calls for internal endpoints (core and plugin APIs)

Reply:
> How will we do calls for internal endpoints (core and plugin APIs)

We'll fetch the target host(s) from the nodes in the cluster.  We'll pass along the same authentication header that the user sent with their original request.  If there are special considerations (Sigv4, etc.) we'll handle those similarly to how they're being handled with the ML Commons use cases.